main/src/ca/uwaterloo/flix/language/errors/WeederError.scala

```diff
@@ -864,4 +864,27 @@ object WeederError {
          |""".stripMargin
     })
   }
+
+  case class IllegalJavaClass(name: String, loc: SourceLocation) extends WeederError {
+    def summary: String = "${name} is not a valid Flix identifer."
+
+    def message(formatter: Formatter): String = {
+      import formatter._
+      s"""${line(kind, source.name)}
+         |>> ${red(name)} is not a valid Flix identifer.
+         |
+         |${code(loc, "illegal identifer")}
```

Comment:
I think we're avoiding the word 'illegal' to sound more friendly. Not sure what the best replacement is though.